proposals/0000-local-modules.rst

```diff
@@ -194,143 +195,294 @@ Motivation
    ``import module`` declaration cannot induce a dependency.
 
+   If the declaration brings no new identifiers into scope (because, for example,
+   the module name is not used to qualify any identifiers), it is an error.
+
    This declaration form is allowed only with ``-XLocalModules``.
 
-10. Local modules may be imported with this import item::
+#. Local modules may be exported with this export item::
+
+     export ::= ... | 'module' [ 'qualified' ] modid [ exports ]
+
+   The meaning of this export item depends on the presence of the ``qualified``
+   keyword in the export item.
+
+   A. With ``qualified``: This exports an export-module containing all names
+      in scope with the *modid*\ ``.`` qualification. If qualified names are in scope
+      further qualified by *modid*\ ``.``, export-modules for those further qualifications
+      are added to the export-module for *modid*. If an *exports* is specified, only those entities
+      are exported. Names in the *exports* can be written unqualified.
+
+      The new behavior is allowed only with ``-XLocalModules``.
+
+   #. Without ``qualified``: As specified in the `Haskell Report <https://www.haskell.org/onlinereport/haskell2010/haskellch5.html#x11-1000005.2>`_,
+      point (5), this also exports (unqualified) all identifiers in scope both with
+      and without the *modid*\ ``.`` prefix. If an *exports* is included, then
+      only those identifiers are included.
 
-      import ::= ... | [ 'import' ] 'module' modid [ impspec ]
+Optional Change Specifications
+------------------------------
 
-    This brings the local module named *modid* into scope. It is an error if
-    this import item appears in a list of imports from a module that does not
-    export the local module *modid*.
+Each numbered item in this section can be considered separately.
 
-    Also brought into scope are the entities listed in the *impspec*;
-    these are brought into scope qualified with the *modid*\ ``.`` prefix. If
-    the *impspec* is not provided, then all entities in *modid* are
-    brought into scope (qualified).
+1. A local module declaration can omit the module name,
+   making an anonymous local module. The names exported
+   by an anonymous module are not added to the enclosing environment qualified,
+   as there is no name to qualify by.
 
-    If the ``import`` keyword is included, then all entities brought into
-    scope qualified are also brought into scope unqualified.
+   New BNF::
 
-    This new form of import is allowed only with ``-XLocalModules``.
+     topdecl ::= ... | 'module' [ modid ] [ exports ] 'where' decls
 
-11. Local modules may be exported with this export item::
+   It is an error to specify a non-empty export list, omit a module's name, and
+   not include the ``import`` keyword. (Anonymous modules without an
+   non-empty export list but without the ``import`` keyword are still useful
+   as a way of declaring instances that use local definitions.)
```

Comment:
I guess “not include the `import` keyword”, here, should be “include the `qualified` keyword” instead.

Should “specify a non-empty export list” be “not specify an empty export list”?

Reply:
Good point.